tests/50federation/33room-get-missing-events.pl

```diff
@@ -484,8 +484,12 @@ sub sytest_user_and_room_fixture {
             assert_json_list( my $earliest = $body->{earliest_events} );
             @$earliest == 1 or
                die "Expected a single 'earliest_event' ID";
-            assert_eq( $earliest->[0], $room->id_for_event( $latest_event ),
-               'earliest_events[0]' );
+            # It is expected that the earliest event is the m.room.member event,
+            # but it is possible that the caches have not yet been invalidated
+            # so also allow any of that event's previous events.
+            my @expected = @{$latest_event->{prev_events}};
+            push( @expected, $room->id_for_event( $latest_event ) );
+            assert_in_list( $earliest->[0], @expected, 'earliest_events[0]' );
```

Comment:
I think this is going to use `'earliest_events[0]'` as one of the things that `$earliest->[0]` is allowed to match, which isn't quite what you want.

I'd just do: 

```perl
assert_ok( any { $earliest->[0] eq $_ } @expected, "'earliest_events' did not match" );
```

but if you prefer a utility function that's ok; just call it properly :)

Reply:
Ah, that's a bit easier. :)